controls/management/commands/exportcomponentlibrary.py

```diff
@@ -0,0 +1,100 @@
+import sys
+import os.path
+
+from django.core.management import call_command
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction, models
+from django.db.utils import OperationalError
+from django.conf import settings
+from pathlib import Path
+from pathlib import PurePath
+from django.utils.text import slugify
+
+# from siteapp.models import User, Organization, Portfolio
+from controls.models import Element, Statement
+# from controls.views import system_element_download_oscal_json
+from controls.views import OSCALComponentSerializer
+
+import fs, fs.errors
+
+
+class Command(BaseCommand):
+    help = 'Export all components.'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--format', metavar='format', nargs='?', default="oscal", help="File format")
+        parser.add_argument('--path', metavar='dir_or_pdf', nargs='?', default="local/export/components", help="The directory path to write export file(s) into.")
+
+    def handle(self, *args, **options):
+
+        # Configure
+        FORMAT = options['format']
+        EXPORT_PATH = options['path']
+        # Create export directory path
+        if not os.path.exists(EXPORT_PATH):
+            Path(EXPORT_PATH).mkdir(parents=True, exist_ok=True)
+
+        # Export the component library
+        elements = Element.objects.filter(element_type="system_element")
+        element_cnt = len(elements)
+
+        if FORMAT == 'oscal':
+            counter = 0
+            for element in elements:
+                counter += 1
+                print(f"{counter} id: {element.id}, element: {element.name}")
+                # Get the impl_smts for component
+                impl_smts = Statement.objects.filter(producer_element=element)
+                filename = str(PurePath(slugify(element.name)).with_suffix('.json'))
+                body = OSCALComponentSerializer(element, impl_smts).as_json()
+                # Save component OSCAL
+                with open(os.path.join(EXPORT_PATH,filename), "w") as f:
+                    f.write(body)
```

Comment:
If it's json. I'd suggest doing for a cleaner inspection of raw data if need to debug later: 
json.dump(body, f, sort_keys=True, indent=4)

Reply:
I think the OSCALComponentSerializer is doing the nice formatting of the JSON output.
The files I am seeing are properly formatted, indented.

snippet of exported file:

```json
{
  "component-definition": {
    "uuid": "77dc0044-3d73-4022-917b-e984de980555",
    "metadata": {
      "title": "Automated Reporting and Cybersecurity Services (demo) Component-to-Control Narratives",
      "published": "2021-05-19T00:33:22+00:00",
      "last-modified": "2021-04-21T17:23:29+00:00",
      "version": "string",
      "oscal-version": "1.0.0-rc1",
      "props": [
        {
          "name": "tag",
          "ns": "https://govready.com/ns/oscal",
          "value": "demo_data"
        },
        {
          "name": "tag",
          "ns": "https://govready.com/ns/oscal",
          "value": "lorem"
        },
        {
          "name": "tag",
          "ns": "https://govready.com/ns/oscal",
          "value": "CaaS"
        }
      ]
    },
    "components": {
      "d885a146-5c84-4459-8869-843eff8ba51b": {
        "title": "Automated Reporting and Cybersecurity Services (demo) (CaaS)",
        "type": "system_element",
        "description": "Automated Reporting and Cybersecurity Services (demo) ",
        "control-implementations": [
          {
            "uuid": "093aea92-6b0f-4b8f-b460-612d9169c7c0",
            "source": "NIST_SP-800-53_rev4",
            "description": "Partial implementation of NIST_SP-800-53_rev4",
            "implemented-requirements": [
              {
                "uuid": "cb3dda05-d69e-4bfe-8621-146e7f80fdc6",
                "control-id": "au-12",
                "description": "",
                "remarks": "",
                "statements": {
                  "au-12_smt": {
                    "uuid": "b660e9eb-ef2f-4fa6-a09f-be8c21cb96c3",
                    "description": "A ipsum perferendis recusandae iure neque, necessitatibus facere exercitationem aliquid debitis omnis ab amet quidem ea dignissimos? Laboriosam asperiores quas modi velit suscipit repellat commodi, dolorem tempora cupiditate voluptatem, optio mollitia unde natus, nesciunt quisquam reprehenderit mollitia delectus natus, nemo impedit laudantium nesciunt?"
                  }
                }
              },
              {
                "uuid": "376c3e4d-3b35-4c53-948c-d0007dbd38b4",
                "control-id": "au-16",
                "description": "",
                "remarks": "",
                "statements": {
                  "au-16_smt.02": {
                    "uuid": "d0745e75-38fd-4a1b-b753-654f6f93cb2f",
                    "description": "Veniam sint illum eveniet sed id, quisquam expedita iusto praesentium officia."
                  }
                }
              },
              {
                "uuid": "7d76531d-4e12-4184-9e80-24a1df58ab12",
                "control-id": "au-2",
                "description": "",
                "remarks": "",
                "statements": {
                  "au-2_smt": {
                    "uuid": "32491bc6-bcd5-44d2-b07f-4edaf8acbdbd",
                    "description": "Ratione inventore rem, harum a voluptas maxime nostrum, distinctio nisi voluptates debitis laudantium nostrum, veritatis tempore incidunt molestias, delectus explicabo quas ipsa deserunt iste perferendis nesciunt animi fugit. Ea necessitatibus hic officia velit, quod inventore nobis illo molestias numquam sint voluptatibus harum exercitationem, repellendus rerum iusto, id assumenda error ipsa ab earum unde delectus possimus dignissimos, maxime modi obcaecati praesentium? Iste voluptas ducimus corrupti ipsa sunt eum aspernatur iusto dicta facere laudantium, excepturi assumenda aliquam autem, rerum deserunt illum nulla ratione maxime, explicabo ipsa minima ratione."
                  },
                  "au-2_smt.03": {
                    "uuid": "14850503-ef63-441c-8633-c3ed9eb01c42",
                    "description": "Nihil consequuntur cum harum possimus veniam reiciendis tempora dolores doloremque, minus ullam ipsam eligendi perspiciatis ad non, labore tempora in praesentium voluptas nesciunt esse ullam dolor obcaecati suscipit veniam, quisquam incidunt alias impedit quia voluptate ab necessitatibus labore sed fuga? Est quibusdam amet quasi in rem saepe ea doloremque fuga eligendi mollitia, sint iste unde quas voluptatibus accusamus veritatis nesciunt, expedita magni enim obcaecati perspiciatis voluptatum excepturi, fuga laborum aperiam vero neque perspiciatis, facilis possimus eum repellendus amet aut eius."
                  }
                }
              },
    ...
    ```